sql/core/src/main/scala/org/apache/spark/sql/execution/SQLExecution.scala

```diff
@@ -71,13 +74,22 @@ object SQLExecution {
       // streaming queries would give us call site like "run at <unknown>:0"
       val callSite = sc.getCallSite()
 
+      val truncateLength = sc.conf.get(SQL_EVENT_TRUNCATE_LENGTH)
+
+      val desc = Option(sc.getLocalProperty(SparkContext.SPARK_JOB_DESCRIPTION))
+        .filter(_ => truncateLength > 0)
+        .map { sqlStr =>
+          Utils.redact(sparkSession.sessionState.conf.stringRedactionPattern, sqlStr)
+            .substring(0, Math.min(truncateLength, sqlStr.length))
```

Comment:
Ur, this line is unsafe because it will throw `IndexOutOfBoundsException` if redacted string is smaller than the original. We need to the length of redacted string to call `substring`.

Reply:
Yes, agreed. Done